Wikimate.php

```diff
@@ -858,3 +858,501 @@ private function findSection($section)
+	 */
+	public function getInfo($refresh = false)
+	{
+		if ($refresh) { // We want to query the API
+
+			$data = array(
+				'titles' => 'File:' . $this->filename,
+				'prop' => 'info|imageinfo',
+				'iiprop' => 'bitdepth|canonicaltitle|comment|parsedcomment|'
+				          . 'commonmetadata|metadata|extmetadata|mediatype|'
+				          . 'mime|thumbmime|sha1|size|timestamp|url|user|userid',
+				'intoken' => 'edit',
+			);
+
+			$r = $this->wikimate->query($data); // Run the query
+
+			// Check for errors
+			if (isset($r['error'])) {
+				$this->error = $r['error']; // Set the error if there was one
+				return null;
+			} else {
+				$this->error = null; // Reset the error status
+			}
+
+			// Get the page (there should only be one)
+			$page = array_pop($r['query']['pages']);
+			unset($r, $data);
+
+			// Abort if invalid file title
+			if (isset($page['invalid'])) {
+				$this->invalid = true;
+				return null;
+			}
+
+			$this->edittoken = $page['edittoken'];
+
+			if (!isset($page['missing'])) {
+				// Update the existance if the file is there
+				$this->exists = true;
+				// Put the content into info
+				$this->info   = $page['imageinfo'][0];
+			}
+			unset($page);
+		}
+
+		return $this->info; // Return the info in any case
+	}
+
+	/**
+	 * Returns the aspect ratio of this image,
+	 * or 0 if file has no image dimensions.
+	 *
+	 * @return  float  The aspect ratio of this image, or 0 if no dimensions
+	 */
+	public function getAspectRatio()
+	{
+		if ($this->info['height'] > 0) {
+			return $this->info['width'] / $this->info['height'];
+		} else {
+			return 0;
+		}
+	}
+
+	/**
+	 * Returns the bit depth of this file.
+	 *
+	 * @return  int  The bit depth of this file
+	 */
+	public function getBitDepth()
+	{
+		return (int)$this->info['bitdepth'];
+	}
+
+	/**
+	 * Returns the canonical title of this file.
+	 *
+	 * @return  string  The canonical title of this file
+	 */
+	public function getCanonicalTitle()
+	{
+		return $this->info['canonicaltitle'];
+	}
+
+	/**
+	 * Returns the comment of this file.
+	 *
+	 * @return  string  The comment of this file
+	 */
+	public function getComment()
+	{
+		return $this->info['comment'];
+	}
+
+	/**
+	 * Returns the common metadata of this file.
+	 *
+	 * @return  array  The common metadata of this file
+	 */
+	public function getCommonMetadata()
+	{
+		return $this->info['commonmetadata'];
+	}
+
+	/**
+	 * Returns the description URL of this file.
+	 *
+	 * @return  string  The description URL of this file
+	 */
+	public function getDescriptionUrl()
+	{
+		return $this->info['descriptionurl'];
+	}
+
+	/**
+	 * Returns the extended metadata of this file.
+	 *
+	 * @return  array  The extended metadata of this file
+	 */
+	public function getExtendedMetadata()
+	{
+		return $this->info['extmetadata'];
+	}
+
+	/**
+	 * Returns the height of this file.
+	 *
+	 * @return  int  The height of this file
+	 */
+	public function getHeight()
```

Comment:
I guess I'd only put width, height and aspect ratio together, since they're all closely related. But there isn't a clear obvious order for the rest of the properties, so I guess it's better to forgo the sorting / grouping of those three.